build-info-extractor-npm/src/test/java/org/jfrog/build/extractor/npm/extractor/NpmExtractorTest.java

```diff
@@ -174,6 +174,49 @@ public void npmInstallTest(Project project, Set<String> expectedDependencies, St
         }
     }
 
+    @DataProvider
+    private Object[][] npmCiProvider() {
+        return new Object[][]{
+                {Project.A, Project.A.dependencies, "", true},
+                {Project.A, Collections.emptySet(), "--only=dev", false},
+                {Project.B, Project.B.dependencies, "", true},
+                {Project.B, Collections.emptySet(), "--production", false},
+                {Project.C, Project.C.dependencies, "", true},
+                {Project.C, Project.A.dependencies, "--only=production", true}
+        };
+    }
+
+    @SuppressWarnings("unused")
+    @Test(dataProvider = "npmCiProvider")
+    public void npmCiTest(Project project, Set<String> expectedDependencies, String args, boolean packageJsonPath) {
```

Comment:
This test is very similar to the ```npmInstallTest``` test. Let's reuse some of the code.

Reply:
Changed both tests to call a shared method which executes the actual build and validates it.